### PR TITLE
Added Support for Virtual  MFA Audit 

### PR DESCRIPTION
**Pull request by Nirbhay1997**
Signed-off-by: nirbhay1997 <nkumar@progress.com>

### Description

Added Support to audit Virtual MFA Device/Devices. This resource does not have a plural resource. No terraform tests were added.

### Check List
Please fill a box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_iam_virtual_mfa_devices.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+Use `should_not` to test the entity does not exist.
+
+    describe aws_iam_virtual_mfa_devices do
+      it { should_not exist }
+    end
+
+### be_available
+
+Use `should` to check if the work_group name is available.
```
Comment (soumyo13) [inline docs/resources/aws_iam_virtual_mfa_devices.md:71]:
```suggestion
Use `should` to check if the entity is available.
```

File: libraries/aws_iam_virtual_mfa_device.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
```
Comment (soumyo13) [inline libraries/aws_iam_virtual_mfa_device.rb:1]: Remove the duplicate

File: libraries/aws_iam_virtual_mfa_devices.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  FilterTable.create
+             .register_column(:serial_number, field: :serial_number)
+             .register_column(:path,          field: :path)
+             .register_column(:user_names,     field: :user_name)
+             .register_column(:user_ids,       field: :user_id)
+             .register_column(:arns,           field: :arn)
+             .register_column(:tags,          field: :tags)
+             .register_column(:enable_date,   field: :enable_date)
+             .register_column(:create_date,   field: :create_date)
+             .install_filter_methods_on_resource(self, :table)
```
Comment (soumyo13) [inline libraries/aws_iam_virtual_mfa_devices.rb:26]:
```suggestion
  FilterTable.create
             .register_column(:serial_numbers, field: :serial_number)
             .register_column(:paths,          field: :path)
             .register_column(:user_names,     field: :user_name)
             .register_column(:user_ids,       field: :user_id)
             .register_column(:arns,           field: :arn)
             .register_column(:tags,          field: :tags)
             .register_column(:enable_dates,   field: :enable_date)
             .register_column(:create_dates,   field: :create_date)
             .install_filter_methods_on_resource(self, :table)
```

File: test/integration/verify/controls/aws_iam_virtual_mfa_devices.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+describe aws_iam_virtual_mfa_devices do
```
Comment (soumyo13) [inline test/integration/verify/controls/aws_iam_virtual_mfa_devices.rb:2]: Add the control block and skip the tests.

File: test/unit/resources/aws_iam_virtual_mfa_devices_test.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    user[:user_name] = 'test1'
+    user[:arn] = 'test1'
+    mock_data[:user] = user
+    user[:create_date] = Time.parse("2013-06-11T23:52:02Z2020-06-05T11:30:39.730000+01:00")
+    data[:data] = { :virtual_mfa_devices => [mock_data] }
+    data[:client] = Aws::IAM::Client
+    @iam_client = AWSIAMVirtualMFADevices.new(client_args: { stub_responses: true }, stub_data: [data])
+  end
+
+  def test_iam_client_exists
```
Comment (soumyo13) [inline test/unit/resources/aws_iam_virtual_mfa_devices_test.rb:39]:
```suggestion
  def test_iam_client__mfa_devices_exists
```

File: test/unit/resources/aws_iam_virtual_mfa_devices_test.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    assert @iam_client.exist?
+  end
+
+  def test_paths
+    assert_equal(@iam_client.path, ['/'])
```
Comment (soumyo13) [inline test/unit/resources/aws_iam_virtual_mfa_devices_test.rb:44]:
```suggestion
    assert_equal(@iam_client.paths, ['/'])
```

File: test/unit/resources/aws_iam_virtual_mfa_devices_test.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def test_instance_user_id
+    assert_equal(@iam_client.user_ids, ['test1'])
+  end
+
+  def test_instance_serial_number
+    assert_equal(@iam_client.serial_number, ['test1'])
```
Comment (soumyo13) [inline test/unit/resources/aws_iam_virtual_mfa_devices_test.rb:52]:
```suggestion
    assert_equal(@iam_client.serial_numbers, ['test1'])
```

File: test/unit/resources/aws_iam_virtual_mfa_devices_test.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    user[:path] = '/'
+    mock_data[:serial_number]= "test1"
+    user[:user_id] = 'test1'
+    user[:user_name] = 'test1'
+    user[:arn] = 'test1'
+    mock_data[:user] = user
+    user[:create_date] = Time.parse("2013-06-11T23:52:02Z2020-06-05T11:30:39.730000+01:00")
+    data[:data] = { :virtual_mfa_devices => [mock_data] }
+    data[:client] = Aws::IAM::Client
+    @iam_client = AWSIAMVirtualMFADevices.new(client_args: { stub_responses: true }, stub_data: [data])
```
Comment (soumyo13) [inline test/unit/resources/aws_iam_virtual_mfa_devices_test.rb:36]:
```suggestion
    @response = AWSIAMVirtualMFADevices.new(client_args: { stub_responses: true }, stub_data: [data])
```

**Review by soumyo13 — CHANGES_REQUESTED** [review]
LGTM. Please make the following changes. And also change the description of the PR. If the singular resource is not required please remove it.

**Review by soumyo13 — APPROVED** [review]

**Review by IanMadd — APPROVED** [review]

**Pull request merged**